spring-boot-jta-jpa/src/main/resources/spring-camel.xml

```diff
@@ -0,0 +1,113 @@
+        </property>
+        <property name="jpaVendorAdapter">
+            <bean class="org.springframework.orm.jpa.vendor.HibernateJpaVendorAdapter">
+                <property name="databasePlatform" value="org.hibernate.dialect.MySQLDialect"/>
+            </bean>
+        </property>
+        <property name="jpaProperties">
+            <props>
+                <prop key="hibernate.hbm2ddl.auto">none</prop>
+                <prop key="hibernate.id.new_generator_mappings">false</prop>
+                <prop key="hibernate.transaction.coordinator_class">jta</prop>
+                <prop key="hibernate.transaction.jta.platform">
+                    org.hibernate.engine.transaction.jta.platform.internal.JBossStandAloneJtaPlatform
+                </prop>
+                <prop key="hibernate.show_sql">false</prop>
+                <prop key="hibernate.format_sql">true"</prop>
+            </props>
+        </property>
+    </bean>
+
+    <bean id="jpa" class="org.apache.camel.component.jpa.JpaComponent">
+        <property name="transactionManager" ref="jtaTransactionManager"/>
+        <property name="entityManagerFactory" ref="entityManagerFactory"/>
+        <property name="joinTransaction" value="true"/>
+        <property name="sharedEntityManager" value="false"/>
+    </bean>
+
+    <bean id="jtaConnectionFactory" class="org.apache.activemq.artemis.jms.client.ActiveMQXAConnectionFactory">
+        <constructor-arg value="tcp://localhost:61616"/>
+        <constructor-arg value="admin"/>
+        <constructor-arg value="admin"/>
+    </bean>
+
+    <!-- Pooled JTA aware connection factory -->
+    <bean id="pooledJtaConnectionFactory" class="org.messaginghub.pooled.jms.JmsPoolXAConnectionFactory">
+        <property name="transactionManager" ref="transactionManagerImple"/>
+        <property name="connectionFactory" ref="jtaConnectionFactory"/>
+        <property name="maxConnections" value="1"/>
+        <property name="maxSessionsPerConnection" value="100"/>
+        <property name="useAnonymousProducers" value="false"/>
+    </bean>
+
+    <bean id="jms" class="org.apache.camel.component.jms.JmsComponent">
+        <property name="configuration">
+            <bean class="org.apache.camel.component.jms.JmsConfiguration">
+                <property name="transactionManager" ref="jtaTransactionManager"/>
+                <property name="connectionFactory" ref="pooledJtaConnectionFactory"/>
+                <!-- disable local transactions as JTA TM will take care of enrolling -->
+                <property name="transacted" value="false"/>
+                <!-- caching does not work with distributed transactions -->
+                <property name="cacheLevelName" value="CACHE_NONE"/>
+                <property name="maxConcurrentConsumers" value="1"/>
+                <property name="testConnectionOnStartup" value="true"/>
+            </bean>
+        </property>
+    </bean>
+</beans>
```

Comment:
Its a bit sad or scary that it takes 100+ lines of spring bean configuration to setup JTA with a database and JMS :(

Reply:
Agreed. One of the objective of this example was to make that complexity explicit. It is even more complicated if you look at beans graph resulting from Narayana SB auto-configuration. There additional wrapper and proxy classes.